src/generic/oomph_utilities.cc

```diff
@@ -1306,6 +1306,41 @@ namespace oomph
         return double(t) / double(CLOCKS_PER_SEC);
       }
     }
+
+    /// Returns a nicely formatted string from an input time in seconds;
+    /// the format depends on the size of time, e.g.:
+    ///  3710 will be printed as 1h:01:50
+    ///   700 will be printed as 11m:40
+    ///    59 will be printed as 59s
+    std::string format_time(const double& time)
+    {
+      std::ostringstream ss;
+  
+      unsigned total_time = unsigned(time);
+      unsigned time_hrs = unsigned(total_time) / 3600;
+      unsigned time_min = (unsigned(total_time) % 3600) / 60;
+      unsigned time_sec = (unsigned(total_time) % 3600) % 60;
```

Comment:
Are you sure you need the explicit `unsigned()` casts on the RHS? I don't think you do...

Reply:
good point, total_time is already unsigned. And I can remove that variable entirely once I change the function argument type to unsigned